src/Uno.UWP/UI/ViewManagement/ApplicationView.iOS.cs

```diff
@@ -8,41 +8,46 @@
 
 namespace Windows.UI.ViewManagement
 {
-    partial class ApplicationView
+	partial class ApplicationView
 	{
 		private static bool UseSafeAreaInsets => UIDevice.CurrentDevice.CheckSystemVersion(11, 0);
 
 		internal void SetCoreBounds(UIKit.UIWindow keyWindow, Foundation.Rect windowBounds)
 		{
-            var statusBarHeight = UIApplication.SharedApplication.StatusBarFrame.Size.Height;
+			var inset = UseSafeAreaInsets
+					? keyWindow.SafeAreaInsets
+					: new UIEdgeInsets(0, 0, 0, 0);
 
-			UIEdgeInsets inset = new UIEdgeInsets(statusBarHeight, 0, 0, 0);
 			// Not respecting its own documentation. https://developer.apple.com/documentation/uikit/uiview/2891103-safeareainsets?language=objc
-			// iOS returns all zeros for SafeAreaInsets on non-iPhoneX phones. (ignoring nav bars or status bars)
-			// For that reason, we will set the window's visible bounds to the SafeAreaInsets only for iPhones with notches,
-			// other phones will have insets that consider the status bar
-			if (UseSafeAreaInsets)
+			// iOS returns all zeros for SafeAreaInsets on non-iPhones. (ignoring nav bars or status bars)
+			// So we need to update the top inset depending of the status bar visibilty on other devices
+			if (UIDevice.CurrentDevice.UserInterfaceIdiom != UIUserInterfaceIdiom.Phone)
 			{
-				if (keyWindow.SafeAreaInsets != UIEdgeInsets.Zero) // if we have a notch
-				{
-					inset = keyWindow.SafeAreaInsets;
-				}
+				inset.Top = UIApplication.SharedApplication.StatusBarHidden
+					? 0
+					: UIApplication.SharedApplication.StatusBarFrame.Size.Height;
 			}
 
-            VisibleBounds = new Foundation.Rect(
+			var newVisibleBounds = new Foundation.Rect(
 				x: windowBounds.Left + inset.Left,
 				y: windowBounds.Top + inset.Top,
 				width: windowBounds.Width - inset.Right - inset.Left,
 				height: windowBounds.Height - inset.Top - inset.Bottom
 			);
 
-			if(this.Log().IsEnabled(Microsoft.Extensions.Logging.LogLevel.Debug))
+			if (VisibleBounds != newVisibleBounds)
 			{
-				this.Log().Debug($"Updated visible bounds {VisibleBounds}, SafeAreaInsets: {inset}");
-			}
+				VisibleBounds = newVisibleBounds;
+				keyWindow?.RootViewController?.SetNeedsStatusBarAppearanceUpdate();
```

Comment:
What's this needed for?

Reply:
It was for testing. I just forgot to remove this one

Reply:
Ok glad I asked :p